lib/molinillo/resolution.rb

```diff
@@ -185,6 +195,9 @@ def process_topmost_state
           create_conflict if state.is_a? PossibilityState
           unwind_for_conflict until possibility && state.is_a?(DependencyState)
         end
+      rescue CircularDependencyError => root_error
```

Comment:
Does now. Still don't love the way I've implemented rescuing from CircularDependency errors, though. I'll have a think about it once I get properly handling previous conflicts working fast.